exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -141,18 +141,22 @@ def collect(self) -> None:
         if self._callback is not None:
             self._callback()
 
+        metric_family_id_metric_family = {}
+
         while self._metrics_to_export:
             for export_record in self._metrics_to_export.popleft():
```

Comment:
With this change we no longer yield with every iteration of this `for` loop, but we yield only when this `for` loop is over. What if something keeps adding metrics to `self._metrics_to_export` while this `for` loop is running faster than we are able to consume them?

Reply:
I think this should be fine. `self._metrics_to_export` is collection of `Metric` batches, so this for loop is just over a single batch and bound to complete regardless of new batches being added to `_metrics_to_export` queue.